WordPress/WordPressTest/MediaServiceTests.swift

```diff
@@ -0,0 +1,114 @@
+import Foundation
+@testable import WordPress
+
+class MediaServiceTests: XCTestCase {
+    private var contextManager: TestContextManager!
+    private var mediaService: MediaService!
+    private var mediaBuilder: MediaBuilder!
+    private var postBuilder: PostBuilder!
+
+    private var context: NSManagedObjectContext {
+        return contextManager.mainContext
+    }
+
+    override func setUp() {
+        contextManager = TestContextManager()
+        mediaService = MediaService(managedObjectContext: context)
+        mediaBuilder = MediaBuilder(context)
+        postBuilder = PostBuilder(context)
+    }
+
+    // MARK: - Tests for failedMediaForUpload(automatedRetry:)
+
+    /// Requesting media for manual attempt, no previous automated retry failures
+    ///
+    func testThatFailedMediaForUploadWorks() {
+        let media = mediaBuilder.with(remoteStatus: .failed).build()
+
+        let failedMediaForUpload = mediaService.failedMediaForUpload(automatedRetry: false)
+
+        XCTAssertTrue(failedMediaForUpload.contains(media))
+        XCTAssertEqual(failedMediaForUpload.count, 1)
```

Comment:
We've been using Nimble expectations in some tests. I know this is a personal taste, but I found the syntax so much descriptive:

```
expect(failedMediaForUpload).to(contain(media))
expect(failedMediaForUpload.count).to(equal(1))
```

What do you think?

Reply:
Right now we only have around 7 tests using Nimble, vs the rest of the codebase not using it.

I'd say it's more important for me to stay close in this case to our standard.  I'm not sure if there was a discussion around Nimble but it could be good to try and see if there's consensus in using it, first?

For more context on my thoughts: the problem with not following standards is that our codebase becomes more fragmented, harder to maintain and it also makes submitting code harder because it's difficult to know what the preference of the reviewer may be.

Reply:
Ok!